src/Groups/matrices/forms.jl

```diff
@@ -467,3 +468,399 @@ function issingular(f::SesquilinearForm{T}) where T
    f.descr != :quadratic && throw(ArgumentError("The form is not quadratic"))
    return GAP.Globals.IsSingularForm(f.X)
 end
+
+
+
+
+########################################################################
+#
+# From group to form #TODO: there are different approaches. Which is the best?
```

Comment:
I took some experiments. The second approach (using GAP MeatAxe) is the fastest and the cheapest, but it works only for irreducible `G`-modules (so, it is useful to get quickly a form of a group we already know the nature).
The first approach uses more memory than the third one, but a less amount of time.
Anyway, the approaches have different aims. The first one produces all the forms of a certain type preserved by `G` (i.e. a basis for the vector space of all the forms preserved by `G`). The second one returns just one form preserved by `G`. The third one returns some of them, but not all of them.
If I'd choose, I would eliminate at least the third one (using GAP package "forms"), since it is not completely clear to me how it works, and keep the first approach (useful for a group recognition) and the second one (quicker). But it is just my opinion. Maybe you or @ThomasBreuer  have different ideas.

Reply:
Can you explain which benchmarks you performed? Like, which inputs groups did you use (which degree, over which fields?), as depending on that, the outcome of the benchmarks might vary quite a bit...

The approach from the `forms` package is (AFAIK) motivated by constructive group recognition, where solving the full linear system of equations can be too slow for big input groups; while finding *some* invariant forms heuristically usually is enough, and at least in GAP is quite a bit faster, at least if the degree of the matrix group is sufficiently high (I don't recall, but let's say 100x100 and not just 5x5 :-) ). Of course the fact that one of your functions is implemented in Julia with Nemo and the other calls into GAP code also can have a major impact on relative speeds: even if the "forms" algorithm really is asymptotically faster, this may be masked by the Julia code having lower "constants" in their actual performance, and so may be faster for small(er) examples... 

That's why it's important to know what exactly you tested.

Reply:
I took tests in dimension up to 10; never tried in dimension 100.
I tested the three approaches on `GO(1,6,9)`, `GO(9,3)`, `GU(6,5)`, `GO(-1,10,4)`, `Sp(8,5)` and `GU(6,4)`. Should I try some higher dimension, or higher field size?